uaclient/entitlements/__init__.py

```diff
@@ -23,16 +23,27 @@
 ]  # type: List[Type[UAEntitlement]]
 
 
-def entitlement_factory(name: str):
+class EntitlementNotFoundError(Exception):
+    pass
+
+
+def entitlement_factory(name: str, not_found_okay: bool = True):
```

Comment:
I think I am missing something here, but why do we need the `not_found_okay` parameter ?

Reply:
@lucasmoura You're not missing anything. It isn't strictly necessary for the functionality in this PR.

However, because we're adding `enable_by_entitlement_name` to `actions.py`, I wanted it to have defined functionality for what happens when the entitlement doesn't exist. I decided an Error should be raised in that situation. And then I thought it makes more sense if the error is raised in this factory function.

I can move this error to the `enable_by_entitlement_name` function though if you prefer

Reply:
Also open to better parameter names for `not_found_okay` :)

Reply:
But is there a situation where we don't want to raise that error if the entitlement doesn't exist ? That is what I am missing here

Reply:
Ahhh I understand. I'd need to potentially refactor some of the places we call `entitlement_factory`. I think some expect `None` to be returned if the entitlement doesn't exist.

But now that I think more about it, I think that refactor would be worth it. I'll add it as an extra commit if you agree @lucasmoura

Reply:
@lucasmoura refactor is up. We should let jenkins run again because this touched a lot of places